test/bulk_schedule_test.cpp

```diff
@@ -48,3 +49,33 @@ TEST(bulk, bulk_transform) {
         EXPECT_EQ(i, output[i]);
     }
 }
+
+TEST(bulk, cancellation) {
+    unifex::single_thread_context ctx;
+    auto sched = ctx.get_scheduler();
+
+    const std::size_t count = 1000;
+
+    std::vector<int> output(count, 0);
+    const std::size_t compare_index = 3;
+
+    unifex::sync_wait(
+        unifex::bulk_join(
+            unifex::bulk_transform(
+                unifex::bulk_with_stop_source(
+                    unifex::bulk_schedule(sched, count)),
+                [&](std::size_t index, auto& cancel_future_operations) noexcept {
+                    // Stop after third index
+                    if(index == compare_index) {
+                        cancel_future_operations.request_stop();
+                    }
+                    output[index] = index;
+                }, unifex::par_unseq)));
```

Comment:
This should probably be seq to be technically correct given the ordering assumption the example makes.